File: simple.go
// Run with:
// go build simple.go
// sudo sysctl vm.drop_caches=3; ./simple <kjvbible_x100.txt >/dev/null

package main

import (
	"fmt"
	"io"
	"os"
	"sort"
	"strings"
	"time"
)

func main() {
	tStart := time.Now()
	content, err := io.ReadAll(os.Stdin)
	if err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
	tRead := time.Now()

	words := strings.Fields(strings.ToLower(string(content)))
	counts := make(map[string]int)
	for _, word := range words {
		counts[word]++
	}
	tProcess := time.Now()

	var ordered []Count
	for word, count := range counts {
		ordered = append(ordered, Count{word, count})
	}
	sort.Slice(ordered, func(i, j int) bool {
		return ordered[i].Count > ordered[j].Count
	})
	tSort := time.Now()

	for _, count := range ordered {
		fmt.Println(string(count.Word), count.Count)
	}
	tEnd := time.Now()

	fmt.Fprintf(os.Stderr, "Reading   : %v\n", tRead.Sub(tStart).Seconds())
	fmt.Fprintf(os.Stderr, "Processing: %v\n", tProcess.Sub(tRead).Seconds())
	fmt.Fprintf(os.Stderr, "Sorting   : %v\n", tSort.Sub(tProcess).Seconds())
	fmt.Fprintf(os.Stderr, "Outputting: %v\n", tEnd.Sub(tSort).Seconds())
	fmt.Fprintf(os.Stderr, "TOTAL     : %v\n", tEnd.Sub(tStart).Seconds())
}

type Count struct {
	Word  string
	Count int
}

File: simple.py
# Run with the following command to clear the read cache between runs:
#
# sudo sysctl vm.drop_caches=3; python3 simple.py <kjvbible_x100.txt >/dev/null

import collections
import sys
import time

t0 = time.time()
content = sys.stdin.read()
t1 = time.time()
read_time = t1 - t0

counts = collections.Counter(content.lower().split())
t2 = time.time()
process_time = t2 - t1

most_common = counts.most_common()
t3 = time.time()
sort_time = t3 - t2

for word, count in most_common:
	print(word, count)

t4 = time.time()
output_time = t4 - t3

total_time = t4 - t0

print('Reading   :', read_time, file=sys.stderr)
print('Processing:', process_time, file=sys.stderr)
print('Sorting   :', sort_time, file=sys.stderr)
print('Outputting:', output_time, file=sys.stderr)
print('TOTAL     :', total_time, file=sys.stderr)
